tests/test_models.py

```diff
@@ -0,0 +1,1001 @@
+        predefined_model="wide_resnet50_2-kather100K", batch_size=1
+    )
+    # don't run test on GPU
+    output = predictor.predict(
+        dataset, return_probs=True, return_labels=True, on_gpu=False
+    )
+    probs = output["probs"]
+    preds = output["preds"]
+    labels = output["labels"]
+
+    assert len(probs) == len(preds)
+    assert len(probs) == len(labels)
+
+    prob_check = [1.0, 0.9999997615814209]
+    pred_check = [5, 8]
+    for idx, probs_ in enumerate(probs):
+        prob_max = max(probs_)
+        assert (
+            np.abs(prob_max - prob_check[idx]) <= 1e-8 and preds[idx] == pred_check[idx]
+        )
+
+
+def test_patch_predictor_wide_resnet101_2_kather100K():
+    """Test for patch predictor with wide_resnet101_2 on Kather 100K dataset."""
+    file_parent_dir = pathlib.Path(__file__).parent
+    dir_patches = file_parent_dir.joinpath("data/sample_patches/")
+    list_paths = grab_files_from_dir(dir_patches, file_types="*.tif")
+    dataset = Patch_Dataset(list_paths)
+
+    # API 1, also test with return_labels
+    predictor = CNN_Patch_Predictor(
+        predefined_model="wide_resnet101_2-kather100K", batch_size=1
+    )
+    # don't run test on GPU
+    output = predictor.predict(
+        dataset, return_probs=True, return_labels=True, on_gpu=False
+    )
+    probs = output["probs"]
+    preds = output["preds"]
+    labels = output["labels"]
+
+    assert len(probs) == len(preds)
+    assert len(probs) == len(labels)
+
+    prob_check = [1.0, 0.999420166015625]
+    pred_check = [5, 8]
+    for idx, probs_ in enumerate(probs):
+        prob_max = max(probs_)
+        assert (
+            np.abs(prob_max - prob_check[idx]) <= 1e-8 and preds[idx] == pred_check[idx]
+        )
+
+
+def test_patch_predictor_densenet121_kather100K():
+    """Test for patch predictor with densenet121 on Kather 100K dataset."""
+    file_parent_dir = pathlib.Path(__file__).parent
+    dir_patches = file_parent_dir.joinpath("data/sample_patches/")
+    list_paths = grab_files_from_dir(dir_patches, file_types="*.tif")
+    dataset = Patch_Dataset(list_paths)
+
+    # API 1, also test with return_labels
+    predictor = CNN_Patch_Predictor(
+        predefined_model="densenet121-kather100K", batch_size=1
+    )
+    # don't run test on GPU
+    output = predictor.predict(
+        dataset, return_probs=True, return_labels=True, on_gpu=False
+    )
+    probs = output["probs"]
+    preds = output["preds"]
+    labels = output["labels"]
+
+    assert len(probs) == len(preds)
+    assert len(probs) == len(labels)
+
+    prob_check = [1.0, 0.9998136162757874]
+    pred_check = [5, 8]
+    for idx, probs_ in enumerate(probs):
+        prob_max = max(probs_)
+        assert (
+            np.abs(prob_max - prob_check[idx]) <= 1e-8 and preds[idx] == pred_check[idx]
+        )
+
+
+def test_patch_predictor_densenet161_kather100K():
+    """Test for patch predictor with densenet161 on Kather 100K dataset."""
```

Comment:
if you are just changing the pretrained model and weights, then probably best to define a function at the top and call that function in each of these tests for readability. It will reduce the number of lines of code and will be easy to read.

Reply:
Addressed in [this commit](https://github.com/TIA-Lab/tiatoolbox/pull/98/commits/9f821fa9bfe1b4f07acd170a38253a7f472aa790)